### PR TITLE
#89 - Added 404 page to redirect invalid URLs

### LINKED ISSUE #89: General - Create a 404 Page
Every website needs a proper 404 "page not found" page in order to properly inform users whenever they are attempting to navigate to a url / page which does not exist. Create a simple 404 page that just conveys the message that the user is trying to go to an invalid page. Perhaps "404" in big letters in the middle of the page?

### PR DESCRIPTION
**Pull request by TheElbowTurnMaster**
fixes #89

### CODE REVIEW

File: src/pages/PageNotFound.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+export const PageNotFound: React.FC = () => {
+  return (
+    <div
+      style={{
```
Comment (jamescd18) [inline src/pages/PageNotFound.tsx:5]: Every new component we create should have its own folder containing three files:
- the tsx component file
- the css module file for styling
- the test file

This provides the best flexibility for future expansion and ensures we have at least 1 test for every component.

File: src/pages/PageNotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        position: 'absolute',
+        left: '50%',
+        top: '50%',
+        transform: 'translate(-50%, -50%)'
+      }}
```
Comment (jamescd18) [inline src/pages/PageNotFound.tsx:10]: Can you attach a screenshot of what this page looks like when running the app locally?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Great work so far with the basic react component! I left a couple comments, plus the below change. 

As a general policy, you don't want to commit package-lock.json changes unless you're making intentional changes to the package.json too. If you aren't familiar with git revert, check out this (https://www.atlassian.com/git/tutorials/undoing-changes/git-revert) article to learn a bit about it, and revert the package-lock.json changes. That should making the deletions count a lot lower than the current 28k deletions.

File: src/pages/PageNotFound.tsx (hunk 2 above)
Comment (TheElbowTurnMaster) [reply src/pages/PageNotFound.tsx]: image

**Comment by TheElbowTurnMaster** [comment]
The package-lock.json files got changed when I ran npm install. I'm not sure why that happened, but is there a way to prevent that for future reference?

**Comment by jamescd18** [comment]
> The package-lock.json files got changed when I ran npm install. I'm not sure why that happened, but is there a way to prevent that for future reference?

@TheElbowTurnMaster For now, I think the only way to deal with it is to ensure you're not committing those changes along with the rest of your changes. Longer term solutions are definitely something we should look into, but not a high priority.

File: src/pages/page-not-found/page-not-found.module.css [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.page_not_found {
+    text-align: center;
```
Comment (jamescd18) [inline src/pages/page-not-found/page-not-found.module.css:2]: Much better!

File: src/pages/page-not-found/page-not-found.test.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import { PageNotFound } from './page-not-found';
+
+test('404 rendered', () => {
+  render(<PageNotFound />);
+  const title = screen.getByText(/404/i);
```
Comment (jamescd18) [inline src/pages/page-not-found/page-not-found.test.tsx:6]: Perfect! That should be good enough for this component.

File: src/pages/page-not-found/page-not-found.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import styles from './page-not-found.module.css';
+
+export const PageNotFound: React.FC = () => {
+  return (
+    <body>
+      <div className={styles.page_not_found}>
+        <h1 className={styles.h1}>404</h1>
```
Comment (jamescd18) [inline src/pages/page-not-found/page-not-found.tsx:8]: Did you check to see if the styling would still be applied to the h1 and h3 tags if you didn't add the classnames to them?

File: src/pages/page-not-found/page-not-found.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (jamescd18) [inline src/pages/page-not-found/page-not-found.tsx:1]:
```suggestion
```
We're using [React 17](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html), so no need to import react anymore!

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Love the changes! Just a couple quick minor things and it should be good to go!

**Comment by jamescd18** [comment]
Also, feel free to click the "resolve conversation" button on any review comments that you've addressed.

File: src/pages/PageNotFound.tsx (hunk 2 above)
Comment (jamescd18) [reply src/pages/PageNotFound.tsx]: Looks great!

File: src/pages/page-not-found/page-not-found.tsx (hunk 5 above)
Comment (TheElbowTurnMaster) [reply src/pages/page-not-found/page-not-found.tsx:7]: They appear to be necessary

**Review by jamescd18 — APPROVED** [review]
Looks perfect! Thank you so much! @kevinyu328 ready to merge & deploy

**Review by jamescd18 — APPROVED** [review]
Looks great!

**Review by kevinyu328 — APPROVED** [review]
Great work! Thanks!

**Pull request merged**